test/jdk/java/awt/print/PrinterJob/InvalidPage.java

```diff
@@ -21,223 +21,120 @@
  * questions.
  */
 
-/**
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Panel;
+import java.awt.print.PageFormat;
+import java.awt.print.Paper;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+
+import jtreg.SkippedException;
+
+/*
  * @test InvalidPage.java
  * @bug 4671634 6506286
  * @summary Invalid page format can crash win32 JRE
  * @key printer
+ * @library /test/lib /java/awt/regtesthelpers
+ * @build PassFailJFrame jtreg.SkippedException
  * @run main/manual InvalidPage
  */
-
-import java.awt.*;
-import java.awt.event.*;
-import java.awt.print.*;
-
 public class InvalidPage extends Frame implements Printable {
 
-  PrinterJob pJob;
-  PageFormat pf;
-
-  public InvalidPage() {
-    super ("Validate Page Test");
-    pJob = PrinterJob.getPrinterJob();
-    pf = pJob.defaultPage();
-    Paper p = pf.getPaper();
-    p.setImageableArea(0,0,p.getWidth(), p.getHeight());
-    pf.setPaper(p);
-    setLayout(new FlowLayout());
-    Panel panel = new Panel();
-    Button printButton = new Button ("Print");
-    printButton.addActionListener(new ActionListener() {
-                public void actionPerformed (ActionEvent e) {
-                    try {
-                         if (pJob.printDialog()) {
-                             pJob.setPrintable(InvalidPage.this, pf);
-                             pJob.print();
-                    }
-                    } catch (PrinterException pe ) {
-                    }
+    PrinterJob pJob;
+    PageFormat pf;
+
+    public InvalidPage() {
+        super("Validate Page Test");
+        pJob = PrinterJob.getPrinterJob();
+        pf = pJob.defaultPage();
+        Paper p = pf.getPaper();
+        p.setImageableArea(0, 0, p.getWidth(), p.getHeight());
+        pf.setPaper(p);
+        setLayout(new FlowLayout());
+        Panel panel = new Panel();
+        Button printButton = new Button("Print");
+        printButton.addActionListener(e -> {
+            if (pJob.printDialog()) {
+                pJob.setPrintable(InvalidPage.this, pf);
+                try {
+                    pJob.print();
+                } catch (PrinterException pe) {
+                    PassFailJFrame.forceFail("Test Failed");
                 }
```

Comment:
I suggest using `printStackTrace` for debugging purposes.

You may want to include the message from the exception as well as hint that the test fails because `PrinterException` is thrown/caught.